src/components/custom/_eu-redirect-modal.tsx

```diff
@@ -8,6 +7,17 @@ import device from 'themes/device.js'
 // Icons
 import Close from 'images/svg/custom/close-2.svg'
 
+type EURedirectProps = {
+    aria_label?: string
+    closeModal?: () => void
+    is_open: boolean | (() => void)
+    ref?: Ref<HTMLAnchorElement>
+    rel?: string
+    target?: string
+    to?: string
+    toggle: boolean | (() => void)
```

Comment:
`boolean | (() => void)` this is being use quite often. Can you create a type for this under `types/generics` the type name can be `BooleanFuncVoidType`